src/test/java/co/cask/hydrator/python/transform/PythonTransformNativeTest.java

```diff
@@ -0,0 +1,276 @@
+  }
+
+  @Test
+  public void testAddRecords() throws Exception {
+    String script = "def transform(record, emitter, context):\n" +
+      "  emitter.emit(record)\n" +
+      "  record['s'] = 'value'\n" +
+      "  emitter.emit(record)";
+
+    Map<String, String> properties = ImmutableMap.<String, String>of("script", script,
+                                                                      "executionMode", getExecutionMode(),
+                                                                      "pythonBinary", "python");
+
+    List<StructuredRecord> outputRecords = runPythonEvaluatorPipeline(INPUT_DEFAULT, properties);
+
+    ImmutableList expected = ImmutableList.builder().
+      addAll(INPUT_DEFAULT).
+      add(StructuredRecord.builder(SCHEMA)
+          .set("s", "value").set("i", 0).set("l", 0L).set("f", 0f).set("d", 0d).set("b", true).build()).
+      add(StructuredRecord.builder(SCHEMA)
+          .set("s", "value").set("i", -10).set("l", -10L).set("f", -10f).set("d", -10d).set("b", true).build()).
+      add(StructuredRecord.builder(SCHEMA)
+          .set("s", "value").set("i", 10).set("l", 10L).set("f", 10f).set("d", 10d).set("b", false).build()).
+      add(StructuredRecord.builder(SCHEMA)
+          .set("s", "value").set("i", 0).set("l", 0L).set("f", 0f).set("d", 0d).set("b", false).build()).
+      build();
+
+    Assertions.assertThat(outputRecords).containsExactlyInAnyOrderElementsOf(expected);
+  }
+
+  @Test
+  public void testModifyRecords() throws Exception {
+    String script = "def transform(record, emitter, context):\n" +
+      "  record['i'] *= 2\n" +
+      "  emitter.emit(record)";
+
+    Map<String, String> properties = ImmutableMap.<String, String>of("script", script,
+                                                                      "executionMode", getExecutionMode(),
+                                                                      "pythonBinary", "python");
+
+    List<StructuredRecord> outputRecords = runPythonEvaluatorPipeline(INPUT_DEFAULT, properties);
+
+    List<StructuredRecord> expected = ImmutableList.of(
+      StructuredRecord.builder(SCHEMA)
+        .set("s", "ab").set("i", 0).set("l", 0L).set("f", 0f).set("d", 0d).set("b", true).build(),
+      StructuredRecord.builder(SCHEMA)
+        .set("s", "xy").set("i", -20).set("l", -10L).set("f", -10f).set("d", -10d).set("b", true).build(),
+      StructuredRecord.builder(SCHEMA)
+        .set("s", "a").set("i", 20).set("l", 10L).set("f", 10f).set("d", 10d).set("b", false).build(),
+      StructuredRecord.builder(SCHEMA)
+        .set("s", "").set("i", 0).set("l", 0L).set("f", 0f).set("d", 0d).set("b", false).build());
+
+    Assertions.assertThat(outputRecords).containsExactlyInAnyOrderElementsOf(expected);
+  }
+
+
+  @Test
+  public void testImportThirdPartyLibrary() throws Exception {
+    // functionality is not supported in Interpreted mode
+    Assume.assumeTrue(getExecutionMode().equals("Native"));
```

Comment:
Instead of this, it's cleaner to have a common test that contains tests for both native and interpreted. Then have a native one that extends the common that also adds this test, and have an interpreted one that extends common and does not have this test.